src/classes/STG_InstallScript_TEST.cls

```diff
@@ -107,13 +119,89 @@ public with sharing class STG_InstallScript_TEST {
-        Test.testInstall(new STG_InstallScript(), new Version(1,35), true);
+        Test.testInstall(new STG_InstallScript(), new Version(1,36), true);
         Test.stopTest();
 
         // They should now all have the 'hed' as the namespace
         List<Trigger_Handler__c> tdtmHandlers = [select Name, Owned_by_Namespace__c from Trigger_Handler__c WHERE Owned_by_Namespace__c = 'hed'];
 
         System.assertEquals(previousDefaults.size(), tdtmHandlers.size());
+
+        // User handlers should not be touched
+        List<Trigger_Handler__c> userHandlers = [select Name, Owned_by_Namespace__c from Trigger_Handler__c WHERE Class__c = 'Contact_Update' AND Object__c = 'Contact'];
+        System.assertEquals(1, userHandlers.size());
+        System.assertEquals(null, userHandlers[0].Owned_by_Namespace__c);
+    }
+
+    // Testing a 1.36 and 1.37 Org state upgrade. This is testing a scenario where an org had user managed triggers, prior to the existence of the "Owned by Namespace" field
+    @isTest
+    public static void tdtmUpdateWithNamespaceUserManaged() {
+
+        // Simulate a user adding their own handler, should not need User_Managed__c to be true
+        Trigger_Handler__c newHandler = new Trigger_Handler__c(
+            Active__c = true,
+            Asynchronous__c = false,
+            Class__c = 'Contact_Update',
+            Load_Order__c = 0.2,
+            Object__c = 'Contact',
+            Trigger_Action__c = 'BeforeInsert;BeforeUpdate'
+        );
+        insert newHandler;
+
+        // Remove Namespace to simulate pre 1.38 status, and user disabled
+        List<Trigger_Handler__c> userModifiedDefaults = TDTM_DefaultConfig.getDefaultRecords();
+        for(Trigger_Handler__c h:userModifiedDefaults){
+            h.Owned_by_Namespace__c = null;
+            h.Active__c = false;
+            h.User_Managed__c = true;
+        }
+        insert userModifiedDefaults;
+
+        //Run the install script
+        Test.testInstall(new STG_InstallScript(), new Version(1,37), true);
+
+        List<Trigger_Handler__c> tdtmHandlers = [SELECT Id, Name, Owned_by_Namespace__c, Active__c, User_Managed__c, Class__c FROM Trigger_Handler__c];
+
+        // Verify the number of default records didn't change. We subtract one because the custom non-HEDAtriiger handler should still exist
+        System.assertEquals(TDTM_DefaultConfig.getDefaultRecords().size(), tdtmHandlers.size()-1);
+
+        for(Trigger_Handler__c h:tdtmHandlers){
+            if(h.Class__c != 'Contact_Update') { // exclude custom handler
+                System.assertEquals('hed', h.Owned_by_Namespace__c);
+
+                // Verify Admin overrides
+                System.assertEquals(false, h.Active__c);
+                System.assertEquals(true, h.User_Managed__c);
+            }
+        }
+
+        Test.startTest();
+
+        //Run the install script
+        Test.testInstall(new STG_InstallScript(), new Version(1,38), true);
```

Comment:
Is it really useful to run the install script twice in this test? The new code in the install script runs for any version lower than 1.38. What about running the install script just once, but also adding the verification of the custom handler? It would make the test a little easier to understand.